File: modulo4_while/treinos/divisores.py
"""
1. Leia uma lista de números e que para cada número lido, 
escreva o próprio número e a relação de seus
divisores. (flag número = 0).
"""


def main():
    print('****** Divisores de N Números *****')

    # Entrada
    numero = int(input('Favor digite um Número: '))

    while numero != 0:
        mostrar_divisores(numero)
        numero = int(input('Favor digite um Número: '))

    print(f'> Vc digitou {numero} e por isso tchau!')


def mostrar_divisores(numero):
    candidato = numero

    while candidato >= 1:
        if eh_divisor(candidato, numero):
            print(f'> {candidato}')

        # convergencia
        candidato = candidato - 1


def eh_divisor(candidato, numero):
    return numero % candidato == 0


main()

File: modulo4_while/treinos/divisores.js
/*
1. Leia uma lista de números e que para cada número lido, 
escreva o próprio número e a relação de seus
divisores. (flag número = 0).
*/
import { question } from "readline-sync"

function main() {
  console.log("****** Divisores de N Números *****")

  // Entrada
  let numero = Number(question("Favor digite um Número: "))

  while (numero != 0) {
    mostrar_divisores(numero)
    numero = Number(question("Favor digite um Número: "))
  }

  console.log(`> Vc digitou ${numero} e por isso tchau!`)
}

function mostrar_divisores(numero) {
  let candidato = numero

  while (candidato >= 1) {
    if (eh_divisor(candidato, numero)) {
      console.log(`> ${candidato}`)
    }

    // convergencia
    candidato = candidato - 1
  }
}

function eh_divisor(candidato, numero) {
  return numero % candidato == 0
}

main()
